src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -257,34 +377,84 @@ def builtin_neighbors(
+        memlet=dace.Memlet(data=origin_index_node.data, subset="0", debuginfo=di),
         dst_conn="__idx",
     )
-    state.add_edge(shift_tasklet, "__result", data_access_tasklet, field_index, dace.Memlet())
-    # select full shape only in the neighbor-axis dimension
-    field_subset = tuple(
-        f"0:{shape}" if dim == offset_provider.neighbor_axis.value else f"i_{dim}"
-        for dim, shape in zip(sorted(iterator.dimensions), field_desc.shape)
-    )
-    state.add_memlet_path(
-        iterator.field,
-        me,
-        data_access_tasklet,
-        memlet=create_memlet_at(iterator.field.data, field_subset),
-        dst_conn="__field",
-    )
-    state.add_memlet_path(
-        data_access_tasklet,
-        mx,
-        result_access,
-        memlet=dace.Memlet(data=result_name, subset=neighbor_index, debuginfo=di),
-        src_conn="__result",
+    state.add_edge(
+        shift_tasklet,
+        "__result",
+        neighbor_index_node,
+        None,
+        dace.Memlet(data=neighbor_index_name, subset="0"),
     )
 
-    return [ValueExpr(result_access, iterator.dtype)]
+    if lift_node is not None:
+        _visit_lift_in_neighbors_reduction(
+            transformer,
+            lift_node,
+            lift_args,
+            offset_provider,
+            me,
+            mx,
+            neighbor_index_node,
+            result_node,
+        )
+    else:
+        field_index = "__field_idx"
+        if isinstance(offset_provider, NeighborTableOffsetProvider):
+            neighbor_check = f"{field_index} >= 0"
+        elif isinstance(offset_provider, StridedNeighborOffsetProvider):
+            neighbor_check = (
+                f"{field_index} < {field_desc.shape[offset_provider.neighbor_axis.value]}"
+            )
+        else:
+            assert isinstance(offset_provider, Dimension)
+            raise NotImplementedError(
+                "Neighbor reductions for cartesian grids not implemented in DaCe backend."
+            )
+        data_access_tasklet = state.add_tasklet(
+            "data_access",
+            code=f"__result = __field[{field_index}]"
+            + (
+                f" if {neighbor_check} else {transformer.context.reduce_identity.value}"
+                if offset_provider.has_skip_values
+                else ""
+            ),
+            inputs={"__field", field_index},
+            outputs={"__result"},
+            debuginfo=di,
+        )
+        state.add_edge(
+            neighbor_index_node,
+            None,
+            data_access_tasklet,
+            field_index,
+            dace.Memlet(data=neighbor_index_name, subset="0"),
+        )
+        # select full shape only in the neighbor-axis dimension
+        field_subset = tuple(
+            f"0:{shape}" if dim == neighbor_dim else f"i_{dim}"
+            for dim, shape in zip(sorted(iterator.dimensions), field_desc.shape)
```

Comment:
If you sort, i.e. change the order of the dimensions, should then not also the order of the shape changes as well?

Reply:
Right the dimensions of the dace array are already in the correct order.
Maybe you should add a comment or something, but on the other hand, probably not.